addon/components/-private/scroll-indicators/component.js

```diff
@@ -269,6 +317,12 @@ export default Component.extend({
 
   didInsertElement() {
     this._super(...arguments);
+
+    // debounced, runloop-safe version
+    this._updateIndicators = bind(this, () => {
```

Comment:
Why do we need `bind` here?

Reply:
I think that's a leftover, let's nix it.

Reply:
Aha, I forgot this is why:
![image](https://user-images.githubusercontent.com/2594635/111705101-8b244480-8816-11eb-811e-2c8743bd112b.png)

You need to wrap the `scheduleOnce` in a run loop or else the test suite gets mad. You can't get at it from the test itself.

Reply:
Two more questions on this topic:

- Why do we set `this._updateIndicators` inside `didInsertElement` vs. `init`?
- Are there any issues with using an anonymous function vs. a function defined on the component?

Reply:
1) No objection; consider it moved.

2) Not afaik. Anything we move to a component function is ultimately going to have to be wrapped in a `bind` or something that produces an anonymous function anyway. This forward declaration does a lot of heavy lifting:

* Context binding for a scroll listener, two `ResizeSensor`s, and a `MutationObserver`
* Deals with the test runloop situation
* Encapsulates the `scheduleOnce`
* Simplifies removing the scroll listener on teardown; don't have to cache a `this._scrollListener` function

What we _can_ do is dump the `bind` for a `run`. The context is already available via the anonymous closure, so `run` is a more precise solution to its problem and can be documented as such.

Reply:
`run` won't cut it for `ember-2.8-lts` test suite. I've notated the seemingly redundant `bind` for now; we can visit when we drop 2.8 support.